tsig.go

```diff
@@ -162,20 +162,29 @@ func tsigGenerateProvider(m *Msg, provider TsigProvider, requestMAC string, time
 	if err != nil {
 		return nil, "", err
 	}
+
 	buf, err := tsigBuffer(mbuf, rr, requestMAC, timersOnly)
 	if err != nil {
 		return nil, "", err
 	}
 
 	t := new(TSIG)
-	// Copy all TSIG fields except MAC and its size, which are filled using the computed digest.
+	// Copy all TSIG fields except MAC, its size, and time signed which are filled when signing.
 	*t = *rr
-	mac, err := provider.Generate(buf, rr)
-	if err != nil {
-		return nil, "", err
+	t.TimeSigned = 0
+	t.MAC = ""
+	t.MACSize = 0
+
+	if rr.Error == RcodeSuccess || rr.Error == RcodeBadTime {
```

Comment:
If the RFC only wants us to avoid signing on "an error relating to the key or MAC", wouldn't this make more sense as "rr.Error != ..." for the relevant errors? I presume we're still aloud and want to sign other error values.

Reply:
Yeah, that makes better sense if we’re allowed to return other errors.

Reply:
Done. Thanks!